### PR TITLE
Upgrade common library to 1.x

### PR DESCRIPTION
**Pull request by kang-makes**
This chart used the override system we should get rid of.

This PR should upgrade the common library and fix the issue of charts overwriting each other its helpers.

### CODE REVIEW

**Review by marcsanmi — APPROVED** [review]
LGTM. That said, I'd wait for another review of someone with more helm charts expertise.

File: charts/newrelic-k8s-metrics-adapter/templates/_helpers.tpl
Comment (sigilioso) [inline charts/newrelic-k8s-metrics-adapter/templates/_helpers.tpl:13]: Probably I'm missing something, but I'm asking just in case I'm not: wouldn't it work to set those values as defaults in `values.yaml` and directly use `newrelic.common.securityContext.pod` in `deployment.yaml`?

Comment (kang-makes) [reply charts/newrelic-k8s-metrics-adapter/templates/_helpers.tpl:13]: The thing here is that if you put something in the `values.yaml` directly it is used always and any kind of `.global.podSecurityContext` is ignored always.

So we honor what it comes from `global`, then `local` and in case the user did not provide any value for security pod's context, then we use our defaults.

**Review by sigilioso — APPROVED** [review]
Thanks for taking care of the comment and the explanations

**Comment by sigilioso** [comment]
Nit: you may update common.library to 1.0.2

**Review by gsanchezgavier — APPROVED** [review]

**Pull request merged**
